File: src/main/java/每日一题/best_time_to_buy_and_sell_stock.java
package 每日一题;

public class best_time_to_buy_and_sell_stock {
    public int maxProfit(int[] prices) {
        if (prices.length < 2) return 0;
        int res = 0;
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                res += (prices[i] - prices[i - 1]);
            }
        }
        return res;
    }
}

File: src/main/java/五毒/第六周/第一遍/best_time_to_buy_and_sell_stock.go
package "stock1"



func maxProfit(prices []int) int {
	n := len(prices)
	if n < 2 {
		return 0
	}
	dp := make([][]int, n)
	var i int
	for i = 0; i < n; i++ {
		dp[i] = make([]int, 2)
	}

	dp[0][0] = 0
	dp[0][1] = -prices[0]

	for i = 1; i < n; i++ {
		dp[i][0] = max(dp[i-1][0], dp[i-1][1]+prices[i])
		dp[i][1] = max(dp[i-1][1], -prices[i])
	}
    return dp[n-1][0]
}

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}
